File: HW/HW10/HW10.java
import java.util.*;

class TreeNode {
	int data;
	TreeNode left;
	TreeNode right;
	
	TreeNode(){
		
	}
	
	TreeNode(int data){
		this.data = data;
	}
	
	TreeNode(int data, TreeNode left, TreeNode right){
		this.data = data;
		this.left = left;
		this.right = right;
	}
}


public class HW10 {
	public List<List<Integer>> levelOrder(TreeNode root){
		List<List<Integer>> result = new ArrayList<>();
		levelOrderHelper(root, result, 0);
		
		return result;
	}
	
	private void levelOrderHelper(TreeNode node, List<List<Integer>> result, int level) {
		if(node == null) {
			return;
		}
		
		if(level == result.size()) {
			result.add(new ArrayList<>());
		}
		
		result.get(level).add(node.data);
		
		levelOrderHelper(node.left, result, level + 1);
		levelOrderHelper(node.right, result, level + 1);
	}
	
	public static void main(String[] args) {
		TreeNode root = new TreeNode(4);
		root.left = new TreeNode(3);
		root.left.left = new TreeNode(1);
		root.right = new TreeNode(8);
		root.right.left = new TreeNode(5);
		root.right.right = new TreeNode(9);
		
		HW10 run = new HW10();
		
		List<List<Integer>> levelOrderTraversal = run.levelOrder(root);
		
		System.out.println("Level order traversal: ");
		for(List<Integer> level: levelOrderTraversal) {
			System.out.print(level);
		}
		
	}
}

File: HW/HW10/HW10.py
from typing import List

class TreeNode(object):
    def __init__(self, data=None, left=None, right=None):
        self.data = data
        self.left = left
        self.right = right

class HW10:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        result = []
        self.levelOrderHelper(root, result, 0)
        return result
    
    def levelOrderHelper(self, node: TreeNode, result: List[List[int]], level: int):
        if node is None:
            return
        if level == len(result):
            result.append([])
        result[level].append(node.data)
        self.levelOrderHelper(node.left, result, level + 1)
        self.levelOrderHelper(node.right, result, level + 1)

root = TreeNode(4)
root.left = TreeNode(3)
root.left.left = TreeNode(1)
root.right = TreeNode(8)
root.right.left = TreeNode(5)
root.right.right = TreeNode(9)

run = HW10()
levelOrderTraversal = run.levelOrder(root)
print("Level order traversal: ")
for level in levelOrderTraversal:
    print(level)

